docs/SPEC.md

```diff
@@ -480,57 +523,67 @@ A _block_ is a possibly empty sequence of statements within matching brace brack
-An identifier may change value via assignment within the same block.
 
 Flux is lexically scoped using blocks:
 
-1. The scope of an option identifier is the options block.
-2. The scope of a preassigned (non-option) identifier is in the universe block.
-3. The scope of an identifier denoting a variable or function at the top level (outside any function) is the package block.
-4. The scope of the name of an imported package is the file block of the file containing the import declaration.
-5. The scope of an identifier denoting a function argument is the function body.
-6. The scope of a variable assigned inside a function is the innermost containing block.
+1. The scope of a preassigned identifier is in the universe block.
+2. The scope of an identifier denoting a variable, option, or function at the top level (outside any function) is the package block.
+3. The scope of the name of an imported package is the file block of the file containing the import declaration.
+4. The scope of an identifier denoting a function argument is the function body.
+5. The scope of an identifier assigned inside a function is the innermost containing block.
 
-An identifier assigned in a block may be reassigned in an inner block with the exception of option identifiers.
-While the identifier of the inner assignment is in scope, it denotes the entity assigned by the inner assignment.
-
-Option identifiers have default assignments that are automatically defined in the _options block_.
-Because the _options block_ is the top-level block of a Flux program, options are visible/available to any and all other blocks.
-
-The package clause is not a assignment; the package name does not appear in any scope.
+Note that the package clause is not an assignment.
+The package name does not appear in any scope.
 Its purpose is to identify the files belonging to the same package and to specify the default package name for import declarations.
 
-
 [IMPL#247](https://github.com/influxdata/platform/issues/247) Add package/namespace support
 
 #### Variable assignment
 
-A variable assignment creates a variable bound to the identifier and gives it a type and value.
-When the identifier was previously assigned within the same block the identifier now holds the new value.
-An identifier cannot change type within the same block.
-
     VariableAssignment = identifier "=" Expression
 
+A variable assignment creates a variable bound to an identifier and gives it a type and value.
+A variable keeps the same type and value for the remainder of its lifetime.
+An identifier assigned to a variable in a block cannot be reassigned in the same block.
+An identifier can be reassigned or shadowed in an inner block.
+
 Examples:
 
     n = 1
-    n = 2
-    f = 5.4
-    r = z()
+    m = 2
+    x = 5.4
+    f = () => {
+        n = "a"
+        m = "b"
+        return a + b
+    }
+
+#### Option assignment
 
+    OptionAssignment = "option" identifier "=" Expression
```

Comment:
We need to update this grammar to allow for a qualified identifier, i.e. `options alert.state = ...`